eq_translations/survey_schema.py

```diff
@@ -1,245 +1,204 @@
-                self.no_context_placeholder_pointers,
-            ) = self.get_placeholder_pointers()
+        with open(schema_path, encoding="utf8") as schema_file:  # pragma: no cover
+            self.schema = json.load(schema_file)  # pragma: no cover
 
     def save(self, schema_path):
-        with open(schema_path, "w", encoding="utf8") as schema_file:
-            return json.dump(self.schema, schema_file, indent=4)
-
-    @property
-    def pointers(self):
-        return self.no_context_pointers + self.context_pointers
-
-    @property
-    def no_context_pointers(self):
-        return (
-            self.get_core_pointers()
-            + self.get_title_pointers()
-            + self.get_message_pointers()
-            + self.get_list_pointers()
-            + self.no_context_placeholder_pointers
-        )
+        with open(schema_path, "w", encoding="utf8") as schema_file:  # pragma: no cover
+            return json.dump(self.schema, schema_file, indent=4)  # pragma: no cover
 
     @property
-    def context_pointers(self):
-        return self.get_answer_pointers() + self.context_placeholder_pointers
-
-    def get_core_pointers(self):
+    def translatable_strings(self):
         """
-        Iterate the schema and return the pointers found
-        :return:
+        :yield: A pointer dictionary
+        :return: A generator
         """
-        pointers = []
-        for key in SurveySchema.no_context_keys:
-            key_pointers = find_pointers_to(self.schema, key)
-            pointers.extend(key_pointers)
-        return pointers
 
-    def get_placeholder_pointers(self):
-        """
-        Placeholder pointers may have context or may not
-        :return:
-        """
-        found_pointers = find_pointers_to(self.schema, "text")
-        context_placeholder_pointers = []
-        no_context_placeholder_pointers = []
+        for key in self.keys_to_translate:
+            with_context = key in self.keys_with_context
 
-        for pointer in found_pointers:
-            if "/answers/" in pointer:
-                context_placeholder_pointers.append(pointer)
-            else:
-                no_context_placeholder_pointers.append(pointer)
+            for pointer in find_pointers_to(self.schema, key):
+                # The 'list' property is used by both 'contents' and 'when' blocks.
+                if "/when/" in pointer:
+                    continue
 
-        return context_placeholder_pointers, no_context_placeholder_pointers
+                self.total_translatable_strings += 1
 
-    def get_message_pointers(self):
-        """
-        Message pointers need to be iterated as a dict and each key added individually
-        :return:
-        """
-        pointers = []
+                schema_element = resolve_pointer(self.schema, pointer)
+                if isinstance(schema_element, dict):
+                    yield from self._get_pointers_from_dict_element(
```

Comment:
When the naming changes, the methods should be renamed appropriately. For example `_get_pointers_from_dict_element` should become something like `_get_translatable_items_from_dict`.